app/src/modules/settings/routes/extensions/components/extension-item.vue

```diff
@@ -57,31 +101,28 @@ const toggleEnabled = async (extensionType?: ExtensionType) => {
 		</v-list-item-content>
 
 		<v-progress-circular v-if="changingEnabledState" indeterminate />
-		<template v-else-if="extension.schema?.type !== 'bundle'">
-			<v-chip v-if="devMode && isAppExtension" v-tooltip.top="t('enabled_dev_tooltip')" class="state enabled" small>
-				{{ t('enabled') }}
-				<v-icon name="lock" right small />
-			</v-chip>
-			<v-chip v-else class="state" :class="{ enabled: extension.meta.enabled }" small>
-				{{ extension.meta.enabled ? t('enabled') : t('disabled') }}
-			</v-chip>
-			<extension-item-options
-				v-if="!devMode || !isAppExtension"
-				class="options"
-				:name="extension.name"
-				:enabled="extension.meta.enabled"
-				@toggle-enabled="toggleEnabled(extension.schema?.type)"
-			/>
-		</template>
+		<v-chip v-if="isLocked" v-tooltip.top="t('enabled_dev_tooltip')" class="state" :class="status.class" small>
+			{{ status.text }}
+			<v-icon name="lock" right small />
+		</v-chip>
+		<v-chip v-else class="state" :class="status.class" small>
+			{{ status.text }}
+		</v-chip>
+		<extension-item-options
+			v-if="!isLocked"
+			class="options"
+			:type="type"
+			:enabled="extension.meta.enabled"
+			@toggle-enabled="toggleEnabled"
+		/>
 	</v-list-item>
 
-	<v-list v-if="children" class="nested">
+	<v-list v-if="children.length" class="nested" :class="{ partial: isPartialEnabled }">
```

Comment:
Lets be specific in the `v-if` if we're updating it anyway.
```suggestion
	<v-list v-if="children.length > 0" class="nested" :class="{ partial: isPartialEnabled }">
```